text/0000-layer-origin-metadata.md

```diff
@@ -0,0 +1,234 @@
+action taking place shouldn't be too difficult.
+
+One complication of implementing this behavior is that the builder
+lifecycle phase does not necessarily know ahead of time what layers
+will be generated by a buildpack. The builder would have to do some
+work to associate the invocation with the generated layers, with the
+result being the layer origin metadata. Having done that work, the
+layer origin metadata can be merged into `report.toml`.
+
+To be explicit, this features does not care about intermediate layer
+states. We can treat buildpacks as black boxes and leave investigation
+of intermeiate layer states to buildpack developers.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* This adds complexity to the lifecycle.
+
+# Alternatives
+[alternatives]: #alternatives
+
+<!-- - What is the impact of not doing this? -->
+
+There's a big problem right now around reproducibiility, and given
+that build reproducibility is a stated goal, something has to be
+done. Until something is done, reproducibilty is not easy.
+
+<!-- - Why is this proposal the best? -->
+
+
+
+<!-- - What other designs have been considered? -->
+
+You can maybe imagine that **a buildpack itself might record all of
+its layer origin metadata** and contribute that directly to
+`report.toml`. However, closer consideration suggests that **this
+should not be the buildpack's responsibility**. Layer origin metadata
+concerns the identity of layers at the container runtime level (or at
+the registry level), which should be opaque to buildpacks; a buildpack
+does not need to know that it's being run in a container.
+
+## Exposing all layer metadata
+[expose-all-layer-metadata-recap]: #exposing-all-layer-metadata
+
+Before this RFC, I'd written an [RFC to expose all layer metadata in
+the image's labels][expose-all-layer-metadata]. My thinking, at the
+time, was that I might write an RFC that could appeal to the CNB
+maintainers, for a feature that would allow me to solve my problem for
+myself. But, that RFC as-is does not address my problems; for one, it
+doesn't link together a buildpack layer X with whatever other
+buildpack layers were around when layer X was created. Non-launch
+layer metadata is important but it's just one piece of the puzzle. I
+think that the ideal solution would involve knowing the identities of
+layers present during a buildpack's execution, as that would enable us
+to discover that layer's metadata, and that ideally we'd solve this
+problem for anyone who ever might have to reproduce a build.
+
+# Prior Art
+[prior-art]: #prior-art
+
+<!-- Discuss prior art, both the good and bad. -->
+
+See [above][expose-all-layer-metadata-recap].
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+<!-- - What parts of the design do you expect to be resolved before this -->
+<!--   gets merged? -->
+
+* Does this affect how `report.toml` should be spec'd?
+* How can I use `image.previous-image-digest` to find that image's
```

Comment:
I think storing `report.toml` files and associating them with images will have to be a platform concern given that we want to avoid embedding that information in the image itself.